lib/iris/common/metadata.py

```diff
@@ -211,32 +837,78 @@ class CubeMetadata(BaseMetadata):
 
     __slots__ = ()
 
-    @wraps(BaseMetadata.name)
-    def name(self, default=None, token=False):
-        def _check(item):
-            return self.token(item) if token else item
+    @wraps(BaseMetadata.__eq__, assigned=("__doc__",), updated=())
+    @lenient_service
+    def __eq__(self, other):
+        return super().__eq__(other)
 
-        default = self.DEFAULT_NAME if default is None else default
+    def _combine_lenient(self, other):
+        """
+        Perform lenient combination of metadata members for cubes.
 
-        # Defensive enforcement of attributes being a dictionary.
-        if not isinstance(self.attributes, Mapping):
-            try:
-                self.attributes = dict()
-            except AttributeError:
-                emsg = "Invalid '{}.attributes' member, must be a mapping."
-                raise AttributeError(emsg.format(self.__class__.__name__))
+        Args:
 
-        result = (
-            _check(self.standard_name)
-            or _check(self.long_name)
-            or _check(self.var_name)
-            or _check(str(self.attributes.get("STASH", "")))
-            or _check(default)
+        * other (CubeMetadata):
+            The other cube metadata participating in the lenient combination.
+
+        Returns:
+            A list of combined metadata member values.
+
+        """
+        # Perform "strict" combination for "cell_methods".
+        value = (
+            self.cell_methods
+            if self.cell_methods == other.cell_methods
+            else None
```

Comment:
Hmmm it's a good question... again a case where I'm not sure so erred on the side of being strict in lenient mode.

I feel like it kinda, sorta depends on the operation the user is performing, and it's not possible to guess the context at this particular lower level in the code. My naive philosophy here was that if the cell methods are the same then it's pretty obvious to know what to do, otherwise don't even attempt to guess what to do (as it might be half right, or even worse completely wrong) so leave the user to apply their intelligence and experience to the metadata.

As a concrete example, consider adding a 3d cube (`t, y, x`) and a 2d (`y, x`) cube. They both have cell methods, but the 3d cube has two cell methods on the `t` and `y` dimensions, whereas the 2d cube has a single cell method on the `y` dimension. Now the result of the `addition` operation is a 3d cube (thru the wonders of broadcasting), and you may argue that the resultant cube should also have two cell methods - having one cell method would be misleading. In actual fact, the context of the operation lives with the `addition` operation, it can make a reasonable assertion of what the cell methods should be, and as it happens it throws away the cell methods, and I kinda agree with that. Even so, the user can easily add them back, if they care and it makes sense to them.

So, after revisiting this and coming full circle again, I'm back to the same conclusion that what I've proposed in this PR seems reasonable, and a safe stance to take (and don't ask me tomorrow, as I may change my mind :smile:)

Anyways, I'm kinda glad that your finding it "non-obvious" and grappling with the same conundrums that I found myself pondering over :wink:

I think we just need to be a fair mix of reasonable and opinionated here.... not unless I'm missing completely obvious.

I've not idea if this has been of any help, whatsoever to you...